### PR TITLE
Improvement requests 

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, my coding partners @brhanuhailu, @yashodhicy and I have created the final capstone project according to the following requirements:

##### Core features - Required for _every_ team size

1. The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
2. In the navigation panel, the user can see links to:
   - Motorcycles/doctors/classes/items that you selected as a theme.
   - "Reserve" form.
   - "My reservations".
   - "Add motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
   - "Delete motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
3. On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).

4. When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).

   - In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

5. When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

6. Make the app responsive, creating both mobile and desktop versions.

##### Additional features - Required for teams with 3-4 & 5 members (optional for others)

7. When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
   - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.

- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
  - Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
  - The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.

8. When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).

9. Add full documentation for your API.

🌺 Back end pull request link
https://github.com/yashodhicy/Final-capstone-back-end/pull/22

### CODE REVIEW

File: src/components/AddHouse.jsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                  name="location"
+                  value={houseDatas.location}
+                  onChange={handleChange}
+                  className="form-control"
+                  id="houseLocation"
+                  placeholder="Enter the location of the house"
+                />
+              </label>
+            </div>
+
```
Comment (mcrd25) [inline src/components/AddHouse.jsx]: These input/label combinations are repetitive and can be abstracted futher with Subcomponents, array or similar and some sort of mapping function

File: src/components/HouseDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    <td>{house.area}</td>
+                  </tr>
+                  <tr>
+                    <td>Number of Rooms</td>
+                    <td>{house.number_of_rooms}</td>
+                  </tr>
+                  <tr>
+                    <td>Price per day</td>
+                    <td>{house.price} $</td>
+                  </tr>
```
Comment (mcrd25) [inline src/components/HouseDetails.jsx:46]: Could this also be abstracted? What if you wanted to add another column/row in future? [OPTIONAL]

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- You have tests 👍 
- UI is functional 👍 

## Required Changes ♻️
_Check the comments under the review._
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions - Your fonts/weights are off a bit and the general UI could flow better with the original and look cleaner.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- user ought to be able to cancel a reservation
- There are some candidates to make some of these html components reusable.
- Seeing where you can make code better by following best practices: See the following articles
   - https://www.freecodecamp.org/news/best-practices-for-react/
   - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Also, you do not have to implement all requested changes, as this is up to you in the end, but I strongly recommend you implement them._

File: src/components/AddHouse.jsx (hunk 1 above)
Comment (brhanuhailu) [reply src/components/AddHouse.jsx]: @mcrd25 Thank you very much for your constructive comment

File: README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### Setup
+
+- ```git clone https://github.com/yashodhicy/Final-capstone-front-end.git```
+
+- ```cd Final-capstone-front-end```
+
+### Running The App Locally
+
+- Run `npm install`
+
```
Comment (Meltrust) [inline README.md]: - [ ] The command to run the app is missing in this section.  Kindly, add the required command to run the frontend app  `npm run dev` 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Mahamadou-Nouridine,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @Mahamadou-Nouridine,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause




Cheers and Happy coding!👏👏👏


------

**Pull request merged**